securedrop_client/gui/widgets.py

```diff
@@ -2190,14 +2196,23 @@ def __init__(
         file_missing: pyqtBoundSignal,
         index: int,
         container_width: int,
+        export_service: Optional[export.Service] = None,
     ) -> None:
         """
         Given some text and a reference to the controller, make something to display a file.
         """
         super().__init__()
 
         self.controller = controller
-        self._export_device = conversation.ExportDevice(controller, controller.export_thread)
+
+        if export_service is None:
+            # Note that injecting an export service that runs in a separate
+            # thread is greatly encouraged! But it is optional because strictly
+            # speaking it is not a dependency of this FileWidget.
+            export_service = export.Service()
+
+        self._export_device = conversation.ExportDevice(controller, export_service)
```

Comment:
I decided to leave a comment here (controversial, for me) because I would like us to avoid this pattern in the future, but still believe it is an acceptable compromise for the time being. YMMV, please let me know if it does! (See the second part of https://github.com/freedomofpress/securedrop-client/pull/1541#issuecomment-1218628883 for my vision to remedy it).